acme/acme/crypto_util.py

```diff
@@ -166,7 +166,7 @@ def probe_sni(name, host, port=443, timeout=300, # pylint: disable=too-many-argu
             " from {0}:{1}".format(
                 source_address[0],
                 source_address[1]
-            ) if socket_kwargs else ""
+            ) if source_address[0] else ""
```

Comment:
If the caller only specifies the `port` in `(host, port)`, it still has an effect. At least, it does on Linux, I'm not sure about other platforms. 

Perhaps:

```suggestion
            ) if any(source_address) else ""
```

Reply:
Right! Missed this, thanks